src/calendar-backend/Calendar.Api/Controllers/CalendarController.cs

```diff
@@ -0,0 +1,162 @@
+using MongoDB.Driver;
+using System.Collections;
+using System.Security.Claims;
+
+namespace Calendar.Api.Controllers;
+
+[ApiController]
+[Route("v1/api/[controller]")]
+public class CalendarController : ControllerBase
+{
+    private readonly ICalendarService calendarService;
+    private readonly IEventService eventService;
+    private readonly IMapper mapper;
+    private readonly ILogger<CalendarController> logger;
+
+    public CalendarController(ICalendarService calendarService, IEventService eventService, IMapper mapper, ILogger<CalendarController> logger)
+    {
+        this.calendarService = calendarService;
+        this.eventService = eventService;
+        this.mapper = mapper;
+        this.logger = logger;
+    }
+
+    #region UserCalendar
+
+    [HttpPost]
+    // [Authorize(Roles = "editor")]
+    public async Task<ActionResult<UserCalendarDTO>> AddCalendar([FromBody] CreateUserCalendarDTO calendar)
+    {
+        if (calendar == null)
+        {
+            return BadRequest();
+        }
+
+        var result = await calendarService.AddCalendarAsync(mapper.Map<UserCalendar>(calendar));
+
+        return Ok(mapper.Map<UserCalendarDTO>(result));
+    }
+
+    [HttpGet]
+    //[Authorize(Roles = "calendar-viewer,calendar-editor")]
+    [Authorize]
+    public async Task<ActionResult<IEnumerable<UserCalendarDTO>>> GetCalendarsByNames()
+    {
+        var groups = base.HttpContext.User.Claims.FirstOrDefault(c => c.Type == ClaimTypes.GroupSid);
+        var testdata = new List<string>
+        {
+            "TINF21AI"
+        }; // TODO: use groups from claim 
+        var calendar = await calendarService.GetCalendarsByNamesAsync(testdata);
+
+        return Ok(mapper.Map<IEnumerable<UserCalendarDTO>>(calendar));
+    }
+
+    [HttpGet("{calendarId}")]
+    // [Authorize(Roles = "viewer,editor")]
+    public async Task<ActionResult<UserCalendarDTO>> GetCalendarById(string calendarId, [FromQuery] bool includeEvents = false)
+    {
+        var calendar = await calendarService.GetCalendarByIdAsync(calendarId, includeEvents);
+        if (calendar == null)
+            return NotFound();
+        return Ok(mapper.Map<UserCalendarDTO>(calendar));
+    }
+
+    [HttpPut("{calendarId}")]
+    // [Authorize(Roles = "editor")]
+    public async Task<ActionResult<UserCalendarDTO>> EditCalendar(string calendarId, [FromBody] UpdateUserCalendarDTO calendar)
+    {
+        if (calendar == null)
+        {
+            return BadRequest();
+        }
+
+        var result = await calendarService.UpdateCalendarAsync(calendarId, mapper.Map<UserCalendar>(calendar));
+        if (result == null)
+            return NotFound();
+        return Ok(mapper.Map<UserCalendarDTO>(result));
+    }
+
+
+    [HttpDelete("{calendarId}")]
+    // [Authorize(Roles = "editor")]
+    public async Task<ActionResult<bool>> DeleteCalendar(string calendarId)
+    {
+        var success = await calendarService.DeleteCalendarByIdAsync(calendarId);
```

Comment:
Hier bei nicht gefundener ID ein Not Found zurückgeben
![image](https://github.com/mhummel-artiso/lecture-calendar/assets/97592367/5dd6d718-9cfb-4bd2-86e4-b590a35b30c5)
Bisher wird nur false zurückgegeben?

Reply:
soll ja auch ein false zurück gegeben da nichts gelöscht wurde, wegen impotent oder wie das heist, true wenn gelöscht wurde